components/google_sign_in/common/features.h

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_GOOGLE_SIGN_IN_COMMON_FEATURES_H_
+#define BRAVE_COMPONENTS_GOOGLE_SIGN_IN_COMMON_FEATURES_H_
+
+#include "base/feature_list.h"
+
+namespace google_sign_in::features {
```

Comment:
I don't think `google_sign_in` component and namespace is required for this feature. Everything related to google signin permission should just go into `permissions` namespace.

Currently it looks like this thing is for the google sign in support, i.e. the browser somehow interacts with it and gets all required data (oauth token, available scopes, etc.), which has nothing to do with what is actually going on.

Reply:
If not a new component, then where would you suggest the URLLoaderThrottle and util classes go? All the examples I've seen so far i.e. debounce, de-amp, speedreader, searchresultsadrenderer are in new components.

Reply:
`brave/components/permissions/`

> If not a new component, then where would you suggest the URLLoaderThrottle and util classes go? All the examples I've seen so far i.e. debounce, de-amp, speedreader, searchresultsadrenderer are in new components.

`brave/components/permissions/google_sign_in` and use `brave/components/permissions/sources.gni` to add those files to build system.

Reply:
done, can you take a look?